### PR TITLE
[SQL] FOREIGN KEY validation and metadata construction

### PR DESCRIPTION
**Pull request by mihaibudiu**
Is this a user-visible change (yes/no): no

The grammar already accepted `FOREIGN KEY`s in multiple forms, but was ignoring this information.
This PR validates the `PRIMARY KEY`s and `FOREIGN KEY`s and stores the information in the circuit.
Subsequent circuit optimizations will take advantage of this information.

### CODE REVIEW

**Comment by mihaibudiu** [comment]
@gz, check MetadataTests.testSchema() for an example of a JSON metadata containing primary and foreign key information.

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/MetadataTests.java [hunk 1]
```diff
@@ -0,0 +1,536 @@
...
+                          "nullable" : false,
+                          "type" : "INTEGER"
+                        }
+                      }
+                    } ],
+                    "primary_key" : [ "COL3" ],
+                    "materialized" : false,
+                    "foreign_keys" : [ {
+                      "columns" : [ "COL2" ],
+                      "refers" : "S",
```
Comment (gz) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/MetadataTests.java:384]: The S isn't defined?

Comment (mihaibudiu) [reply sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/MetadataTests.java:384]: That's why you get a warning too

**Comment by mihaibudiu** [comment]
How can I run just an integration test locally?

**Pull request merged**